pkgs/by-name/we/weblate/package.nix

```diff
@@ -0,0 +1,193 @@
+    (fetchpatch {
+      url = "https://github.com/WeblateOrg/weblate/commit/355c81c977c59948535a98a35a5c05d7e6909703.patch";
+      sha256 = "sha256-9cgE+KeSsXweb2JUvstMxEehfNVktZm1geAo496EPN0=";
+    })
+    # relax dependency constraint for django-crispy-forms
+    (fetchpatch {
+      url = "https://github.com/WeblateOrg/weblate/commit/7b341c523ed9b3b41ecfbc5c92dd6156992e4f32.patch";
+      sha256 = "sha256-AcWQD+Z1wGX9MT3IFiomN4sncsQ2VOtflo6BLRZbxBc=";
+    })
+  ];
+
+  # relax dependency constraint for mistletoe
+  # https://github.com/WeblateOrg/weblate/commit/50df46a25dda2b7b39de86d4c65ecd7a685f62e6
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace '"mistletoe>=1.3.0,<1.4"' '"mistletoe>=1.3.0,<1.5"'
+  '';
+
+  build-system = with python.pkgs; [ setuptools ];
+
+  nativeBuildInputs = [
+    wrapGAppsNoGuiHook
+    gobject-introspection
+  ];
+
+  # Build static files into a separate output
+  postBuild =
+    let
+      staticSettings = writeText "static_settings.py" ''
+        STATIC_ROOT = os.environ["static"] + "/static"
+        COMPRESS_ENABLED = True
+        COMPRESS_OFFLINE = True
+        COMPRESS_ROOT = os.environ["static"] + "/compressor-cache"
+        # So we don't need postgres dependencies
+        DATABASES = {}
+      '';
+    in
+    ''
+      mkdir $static
+      cat weblate/settings_example.py ${staticSettings} > weblate/settings_static.py
+      export DJANGO_SETTINGS_MODULE="weblate.settings_static"
+      ${python.pythonOnBuildForHost.interpreter} manage.py collectstatic --no-input
+      ${python.pythonOnBuildForHost.interpreter} manage.py compress
+    '';
+
+  dependencies = with python.pkgs; [
+    aeidon
+    ahocorasick-rs
+    borgbackup
+    celery
+    certifi
+    charset-normalizer
+    django-crispy-bootstrap3
+    cryptography
+    cssselect
+    cython
+    diff-match-patch
+    django-appconf
+    django-celery-beat
+    django-compressor
+    django-cors-headers
+    django-crispy-forms
+    django-filter
+    django-redis
+    django
+    djangorestframework
+    filelock
+    fluent-syntax
+    gitpython
+    hiredis
+    html2text
+    iniparse
+    jsonschema
+    lxml
+    misaka
+    mistletoe
+    nh3
+    openpyxl
+    packaging
+    phply
+    pillow
+    pycairo
+    pygments
+    pygobject3
+    pyicumessageformat
+    pyparsing
+    python-dateutil
+    python-redis-lock
+    rapidfuzz
+    redis
+    requests
+    ruamel-yaml
+    sentry-sdk
+    siphashc
+    social-auth-app-django
+    social-auth-core
+    tesserocr
+    translate-toolkit
+    translation-finder
+    user-agents
+    weblate-language-data
+    weblate-schemas
+  ];
+
+  optional-dependencies = {
+    postgres = with python.pkgs; [ psycopg ];
+  };
+
+  GI_TYPELIB_PATH = lib.makeSearchPathOutput "out" "lib/girepository-1.0" [
+    pango
+    harfbuzz
+    librsvg
+    gdk-pixbuf
+    glib
+  ];
+  makeWrapperArgs = [ "--set GI_TYPELIB_PATH \"$GI_TYPELIB_PATH\"" ];
```

Comment:
This is actually redundant, so I removed the use of `wrapGappsNoGuiHook`. We need to specify `GI_TYPELIB_PATH` explicitly, so that we can expose it through `passthru`. That's the only reason. I added a comment to explain this.